PowerShell/ScubaGear/Rego/AADConfig.rego

```diff
@@ -444,15 +444,22 @@ tests contains {
 ManagedDeviceAuth contains CAPolicy.DisplayName if {
     some CAPolicy in input.conditional_access_policies
 
-    Contains(CAPolicy.Conditions.Users.IncludeUsers, "All") == true
-    Contains(CAPolicy.Conditions.Applications.IncludeApplications, "All") == true
-    CAPolicy.State == "enabled"
+    PolicyConditionsMatch(CAPolicy) == true
 
     Conditions := [
         "compliantDevice" in CAPolicy.GrantControls.BuiltInControls,
         "domainJoinedDevice" in CAPolicy.GrantControls.BuiltInControls,
+        count(CAPolicy.GrantControls.BuiltInControls) == 2,
+        CAPolicy.GrantControls.Operator == "OR",
     ]
-    count(FilterArray(Conditions, true)) > 0
+
+    
+    count(FilterArray(Conditions, true)) == 4
```

Comment:
Because you want all the checks to be true, that is an AND check & you should utilize Rego's capability (each new line is treated as a AND comparison). FilterArray is used for an OR comparison